### PR TITLE
feat(desktop): GitHub workflow board parity in kanban pane (KAT-2249)

### PR DESCRIPTION
**Pull request by gannonh**
## Summary
- read GitHub tracker configuration from workspace `WORKFLOW.md` frontmatter in the main process and keep renderer-facing metadata typed and backend-neutral
- normalize GitHub workflow state for both label mode and Projects v2 mode into the existing `WorkflowBoardSnapshot` contract
- keep the same kanban renderer path for Linear and GitHub snapshots while exposing source/mode badges and metadata only at presentation level
- add deterministic Electron parity coverage for GitHub labels mode + projects_v2 mode and include explicit smoke-path proof expectations
- harden edge-case tests for config parsing, pagination, auth fallback, stale/error fallback behavior, and HTTP/GraphQL error mapping

## Validation
- `cd apps/desktop && npx vitest run src/main/__tests__/workflow-config-reader.test.ts src/main/__tests__/github-workflow-client.test.ts src/main/__tests__/workflow-board-service.test.ts src/renderer/lib/__tests__/workflow-board.test.ts`
- `cd apps/desktop && bun run typecheck`
- `cd apps/desktop && npx playwright test e2e/tests/workflow-kanban-github.e2e.ts`
- `cd apps/desktop && bun run test`

Closes KAT-2249

## Summary by CodeRabbit

* **New Features**
  * Introduced GitHub workflow board support alongside Linear, with GitHub labels and projects v2 modes
  * Workflow card titles now function as clickable hyperlinks to external resources when URLs are available
  * Expanded board status display to include GitHub configuration details such as state mode and repository information

<h3>Greptile Summary</h3>

This PR wires up a full GitHub workflow board backend for Kata Desktop, enabling the existing kanban pane to display issues from GitHub label-based workflows and GitHub Projects v2. It introduces three main additions: a custom `WORKFLOW.md` frontmatter reader (`workflow-config-reader.ts`), a `GithubWorkflowClient` that normalizes REST (label mode) and GraphQL (projects_v2 mode) responses into the existing `WorkflowBoardSnapshot` contract, and test-mode fixture snapshots for both GitHub modes. The renderer (`KanbanPane`, `SliceCard`) is extended with minimal changes — a source/mode status badge and conditional URL linking — keeping the shared kanban path intact.

Key highlights:
- The `WorkflowBoardSnapshot` contract is correctly reused for GitHub, with `source.githubStateMode`, `source.repoOwner`, and `source.repoName` added as optional presentation-layer metadata.
- Stale-snapshot and error-fallback logic mirrors the Linear path and is well-tested.
- The `workspaceDir` fixture refactor in `electron.fixture.ts` correctly exposes the workspace path for test-time file writes, though it leaves behind a redundant double-`rmSync` across both fixtures.
- **Primary concern**: `SliceCard.tsx` always renders the \"Show tasks\" collapsible regardless of whether `card.tasks` is non-empty. GitHub cards always have `tasks: []`, so every GitHub card will show a non-functional \"Show tasks\" toggle.
- The REST pagination in `fetchAllRepoIssues` uses a hard 10-page ceiling (1,000 issues) rather than the `Link` header; this is a silent cap that could affect large repos.

<h3>Confidence Score: 3/5</h3>

Safe to merge with the SliceCard task-collapsible fix applied; the empty "Show tasks" toggle on every GitHub card is a visible UX regression.

The core backend logic (config parsing, REST/GraphQL client, snapshot normalization, stale/error fallback) is well-structured and thoroughly tested. The main blocker is the SliceCard UX regression that renders a non-functional "Show tasks" toggle on all GitHub cards due to a missing guard on card.tasks.length. The double-cleanup and REST pagination ceiling are style/design issues that don't break functionality.

apps/desktop/src/renderer/components/kanban/SliceCard.tsx (missing task-count guard), apps/desktop/e2e/fixtures/electron.fixture.ts (redundant cleanup), apps/desktop/src/main/github-workflow-client.ts (pagination ceiling)

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| apps/desktop/src/renderer/components/kanban/SliceCard.tsx | Adds conditional URL link rendering for GitHub cards; however the "Show tasks" collapsible is always rendered even for GitHub cards that always have zero tasks. |
| apps/desktop/src/main/github-workflow-client.ts | New GitHub workflow client that fetches issues via REST (label mode) or GraphQL (projects_v2 mode), normalizing both into the shared WorkflowBoardSnapshot contract; REST pagination is hard-capped at 10 pages (1000 issues). |
| apps/desktop/src/main/workflow-board-service.ts | Significantly updated to route between GitHub and Linear backends based on WORKFLOW.md config; adds test fixtures for both GitHub modes; stale/error fallback is well-handled. |
| apps/desktop/src/main/workflow-config-reader.ts | New minimal custom YAML frontmatter parser for WORKFLOW.md; handles inline comments, quoted values, and basic indentation correctly for the expected input format. |
| apps/desktop/e2e/fixtures/electron.fixture.ts | Extracts workspaceDir as a separate fixture to allow test-time file writes; introduces a redundant double-cleanup where both workspaceDir and electronApp teardowns attempt to delete the same dataDir. |
| apps/desktop/e2e/tests/workflow-kanban-github.e2e.ts | New e2e tests covering GitHub labels mode and projects_v2 mode; the smoke-path test is effectively a documentation placeholder. |
| apps/desktop/src/main/__tests__/github-workflow-client.test.ts | Comprehensive unit tests covering label mode, projects_v2 mode, auth fallback, error codes, and pagination edge cases. |
| apps/desktop/src/main/__tests__/workflow-board-service.test.ts | Well-hardened tests covering stale/error fallback, concurrent dedup, fixture mode selection, and both Linear and GitHub error paths. |
| apps/desktop/src/shared/types.ts | Adds WorkflowTrackerConfig union, GithubWorkflowStateMode, and extends WorkflowBoardSnapshot.source with GitHub-specific metadata fields; type additions are well-structured. |

</details>

</details>

<h3>Sequence Diagram</h3>

```mermaid
sequenceDiagram
    participant WS as Workspace WORKFLOW.md
    participant CR as WorkflowConfigReader
    participant SVC as WorkflowBoardService
    participant GH as GithubWorkflowClient
    participant LIN as LinearWorkflowClient
    participant R as KanbanPane (Renderer)

    R->>SVC: refreshBoard()
    SVC->>CR: readWorkspaceWorkflowTrackerConfig(workspacePath)
    CR->>WS: read WORKFLOW.md (frontmatter)
    WS-->>CR: YAML content
    CR-->>SVC: WorkflowTrackerConfig {kind: github|linear}

    alt KATA_TEST_MODE enabled
        SVC-->>R: selectFixtureForMode() snapshot
    else kind === github
        SVC->>GH: fetchSnapshot({config})
        alt stateMode === labels
            GH->>GH: fetchAllRepoIssues (REST, paginated 1-10 pages)
            GH->>GH: extractStateFromLabels + mapLinearStateToColumnId
        else stateMode === projects_v2
            GH->>GH: resolveProjectsV2Field (GraphQL)
            GH->>GH: fetchProjectsV2Items (GraphQL, cursor paginated)
        end
        GH-->>SVC: WorkflowBoardSnapshot {backend: github}
        SVC-->>R: WorkflowBoardSnapshotResponse
    else kind === linear (default)
        SVC->>LIN: fetchActiveMilestoneSnapshot({projectRef})
        LIN-->>SVC: WorkflowBoardSnapshot {backend: linear}
        SVC-->>R: WorkflowBoardSnapshotResponse
    end

    R->>R: normalizeWorkflowColumns(board)
    R->>R: render KanbanColumn → SliceCard per card
```

<details><summary><h3>Comments Outside Diff (2)</h3></summary>

1. `apps/desktop/src/renderer/components/kanban/SliceCard.tsx`, line 38-46 ([link](https://github.com/gannonh/kata/blob/3d5329b9226a0145d8569bbea09128b438330b2f/apps/desktop/src/renderer/components/kanban/SliceCard.tsx#L38-L46)) 

   <a href="#">P1</a> **"Show tasks" collapsible always renders for taskless GitHub cards**

   GitHub cards are always created with `tasks: []` and `taskCounts: { total: 0, done: 0 }` (see `github-workflow-client.ts` lines 166–168 and 253–255). Because the `<Collapsible>` block has no guard, every GitHub card renders a "Show tasks" toggle that opens to an empty `<TaskList />`. This is a UX regression introduced by routing GitHub cards through this shared component.

   The collapsible section should be conditionally rendered when tasks are actually present:

   

   <details><summary>Prompt To Fix With AI</summary>

   `````markdown
   This is a comment left during a code review.
   Path: apps/desktop/src/renderer/components/kanban/SliceCard.tsx
   Line: 38-46

   Comment:
   **"Show tasks" collapsible always renders for taskless GitHub cards**

   GitHub cards are always created with `tasks: []` and `taskCounts: { total: 0, done: 0 }` (see `github-workflow-client.ts` lines 166–168 and 253–255). Because the `<Collapsible>` block has no guard, every GitHub card renders a "Show tasks" toggle that opens to an empty `<TaskList />`. This is a UX regression introduced by routing GitHub cards through this shared component.

   The collapsible section should be conditionally rendered when tasks are actually present:

   

   How can I resolve this? If you propose a fix, please make it concise.
   `````
   </details>

2. `apps/desktop/e2e/fixtures/electron.fixture.ts`, line 86-134 ([link](https://github.com/gannonh/kata/blob/3d5329b9226a0145d8569bbea09128b438330b2f/apps/desktop/e2e/fixtures/electron.fixture.ts#L86-L134)) 

   <a href="#">P2</a> **Redundant double `rmSync(dataDir)` across two fixtures**

   After this refactor, `dataDir` is deleted twice for every test that uses `electronApp` (directly or via `readyWindow`/`mainWindow`):

   1. `electronApp` teardown (~line 133): `rmSync(dataDir, { recursive: true, force: true })`
   2. `workspaceDir` teardown (~line 83): `rmSync(dataDir, { recursive: true, force: true })`

   Playwright tears down fixtures in reverse dependency order, so `electronApp` runs first. By the time `workspaceDir` cleanup fires, the directory is already gone. The `force: true` flag suppresses the resulting error, so it's safe — but the second call is a no-op and its original intent (workspace file cleanup) is already covered by the first.

   Consider removing the `rmSync` from `electronApp`'s `finally` block since `workspaceDir` now owns the lifecycle of `dataDir`. The comment "race with dying process" was valid when `electronApp` owned `dataDir`, but that ownership has moved.

   <details><summary>Prompt To Fix With AI</summary>

   `````markdown
   This is a comment left during a code review.
   Path: apps/desktop/e2e/fixtures/electron.fixture.ts
   Line: 86-134

   Comment:
   **Redundant double `rmSync(dataDir)` across two fixtures**

   After this refactor, `dataDir` is deleted twice for every test that uses `electronApp` (directly or via `readyWindow`/`mainWindow`):

   1. `electronApp` teardown (~line 133): `rmSync(dataDir, { recursive: true, force: true })`
   2. `workspaceDir` teardown (~line 83): `rmSync(dataDir, { recursive: true, force: true })`

   Playwright tears down fixtures in reverse dependency order, so `electronApp` runs first. By the time `workspaceDir` cleanup fires, the directory is already gone. The `force: true` flag suppresses the resulting error, so it's safe — but the second call is a no-op and its original intent (workspace file cleanup) is already covered by the first.

   Consider removing the `rmSync` from `electronApp`'s `finally` block since `workspaceDir` now owns the lifecycle of `dataDir`. The comment "race with dying process" was valid when `electronApp` owned `dataDir`, but that ownership has moved.

   How can I resolve this? If you propose a fix, please make it concise.
   `````
   </details>

</details>

<details><summary>Prompt To Fix All With AI</summary>

`````markdown
This is a comment left during a code review.
Path: apps/desktop/src/renderer/components/kanban/SliceCard.tsx
Line: 38-46

Comment:
**"Show tasks" collapsible always renders for taskless GitHub cards**

GitHub cards are always created with `tasks: []` and `taskCounts: { total: 0, done: 0 }` (see `github-workflow-client.ts` lines 166–168 and 253–255). Because the `<Collapsible>` block has no guard, every GitHub card renders a "Show tasks" toggle that opens to an empty `<TaskList />`. This is a UX regression introduced by routing GitHub cards through this shared component.

The collapsible section should be conditionally rendered when tasks are actually present:

```suggestion
        {card.tasks.length > 0 && (
          <Collapsible open={isOpen} onOpenChange={setIsOpen}>
            <CollapsibleTrigger className="inline-flex items-center gap-1 text-xs font-medium text-primary hover:underline">
              <ChevronDown className={`size-3 transition-transform ${isOpen ? 'rotate-180' : ''}`} />
              {isOpen ? 'Hide tasks' : 'Show tasks'}
            </CollapsibleTrigger>
            <CollapsibleContent className="pt-2">
              <TaskList tasks={card.tasks} />
            </CollapsibleContent>
          </Collapsible>
        )}
```

How can I resolve this? If you propose a fix, please make it concise.

---

This is a comment left during a code review.
Path: apps/desktop/e2e/fixtures/electron.fixture.ts
Line: 86-134

Comment:
**Redundant double `rmSync(dataDir)` across two fixtures**

After this refactor, `dataDir` is deleted twice for every test that uses `electronApp` (directly or via `readyWindow`/`mainWindow`):

1. `electronApp` teardown (~line 133): `rmSync(dataDir, { recursive: true, force: true })`
2. `workspaceDir` teardown (~line 83): `rmSync(dataDir, { recursive: true, force: true })`

Playwright tears down fixtures in reverse dependency order, so `electronApp` runs first. By the time `workspaceDir` cleanup fires, the directory is already gone. The `force: true` flag suppresses the resulting error, so it's safe — but the second call is a no-op and its original intent (workspace file cleanup) is already covered by the first.

Consider removing the `rmSync` from `electronApp`'s `finally` block since `workspaceDir` now owns the lifecycle of `dataDir`. The comment "race with dying process" was valid when `electronApp` owned `dataDir`, but that ownership has moved.

How can I resolve this? If you propose a fix, please make it concise.

---

This is a comment left during a code review.
Path: apps/desktop/src/main/github-workflow-client.ts
Line: 410-422

Comment:
**REST pagination uses a hard page-number ceiling instead of the Link header**

`fetchAllRepoIssues` iterates pages 1–10, breaking early when a page returns fewer than `PAGE_SIZE` items. This is correct for the common case, but it silently caps at 1,000 issues and will miss items in repos with more than that. The GitHub REST API advertises the next page via a `Link: <url>; rel="next"` response header, which is the idiomatic and reliable stop condition.

The existing GraphQL path in `fetchProjectsV2Items` already follows proper cursor-based pagination. Consider applying the same pattern (or at minimum, documenting the 1,000-issue ceiling with a log warning) so operators with large repos aren't silently surprised.

How can I resolve this? If you propose a fix, please make it concise.

---

This is a comment left during a code review.
Path: apps/desktop/src/main/__tests__/workflow-config-reader.test.ts
Line: 153-156

Comment:
**Redundant `mkdirSync` call after `mkdtempSync`**

`mkdtempSync` already creates the directory; calling `mkdirSync(workspace, { recursive: true })` on the same path immediately after is a no-op.

```suggestion
    const workspace = mkdtempSync(path.join(tmpdir(), 'workflow-config-github-projects-'))
    writeFileSync(
```

How can I resolve this? If you propose a fix, please make it concise.
`````

</details>

<sub>Reviews (1): Last reviewed commit: ["test(desktop): harden github workflow bo..."](https://github.com/gannonh/kata/commit/3d5329b9226a0145d8569bbea09128b438330b2f) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=27335023)</sub>

> Greptile also left **2 inline comments** on this PR.

### CODE REVIEW

File: apps/desktop/src/main/github-workflow-client.ts [hunk 1]
```diff
@@ -0,0 +1,593 @@
...
+    if (!response.ok) {
+      throw new GithubWorkflowClientError(
+        'NETWORK',
+        `GitHub API request failed with status ${response.status}`,
+        response.status,
+      )
+    }
+
+    return (text ? JSON.parse(text) : {}) as T
+  }
```
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/desktop/src/main/github-workflow-client.ts:528]: Addressed in commit a1dbfbbd.

- `readJsonOrThrow` now accepts an optional `defaultEmptyResponse`.
- `restRequest` passes `[]` as the default, so empty REST bodies are treated as an empty issue list for array callers.
- For callers without a default, empty bodies now raise a typed `GithubWorkflowClientError`.
- Added regression coverage in `github-workflow-client.test.ts` (`treats empty REST response body as empty issue list`).

File: apps/desktop/src/main/workflow-board-service.ts [hunk 2]
```diff
@@ -135,18 +249,99 @@ export class WorkflowBoardService {
...
+      const snapshot = this.toErrorSnapshot({
+        nowIso,
+        backend: 'github',
+        projectId: 'github:unknown/unknown',
+        message: trackerResult.error.message,
+        code: trackerResult.error.code,
+      })
+      this.lastSnapshot = snapshot
+      return { success: true, snapshot }
+    }
```
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/desktop/src/main/workflow-board-service.ts:267]: Addressed in commit a1dbfbbd.

`workflow-board-service` no longer hard-codes `github` when `readWorkspaceWorkflowTrackerConfig` returns an error. The backend now defaults to `linear` unless the last snapshot backend was GitHub, and the fallback project id is aligned to that backend. Added assertion coverage in `workflow-board-service.test.ts` for the malformed-workspace path (`backend` is now expected to be `linear`).

File: apps/desktop/src/main/github-workflow-client.ts [hunk 3]
```diff
@@ -0,0 +1,593 @@
...
+        `/repos/${encodeURIComponent(repoOwner)}/${encodeURIComponent(repoName)}/issues?state=open&per_page=${PAGE_SIZE}&page=${page}`,
+      )
+
+      issues.push(...pageItems)
+      if (pageItems.length < PAGE_SIZE) {
+        break
+      }
+    }
+
+    return issues
```
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/desktop/src/main/github-workflow-client.ts]: Addressed in commit a1dbfbbd.

Kept the bounded REST pagination but made the limit explicit and visible:
- introduced `MAX_REST_PAGES`
- documented the `MAX_REST_PAGES * PAGE_SIZE` cap in code
- emit a warning log when the cap is reached so truncation is observable

File: apps/desktop/src/main/__tests__/workflow-config-reader.test.ts [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+
+    const result = await readWorkspaceWorkflowTrackerConfig(workspacePath)
+    expect(result.config).toBeNull()
+    expect(result.error?.code).toBe('UNKNOWN')
+  })
+
+  test('parses github projects v2 mode tracker config', async () => {
+    const workspace = mkdtempSync(path.join(tmpdir(), 'workflow-config-github-projects-'))
+    mkdirSync(workspace, { recursive: true })
+    writeFileSync(
```
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/desktop/src/main/__tests__/workflow-config-reader.test.ts:155]: Addressed in commit a1dbfbbd by removing the redundant `mkdirSync(...)` call in the projects_v2 config parser test (`mkdtempSync` already creates the directory).
